src/types/navigation.ts

```diff
@@ -0,0 +1,46 @@
+import type {
+  CompositeScreenProps,
+  NavigatorScreenParams,
+} from "@react-navigation/native";
+import type { NativeStackScreenProps } from "@react-navigation/native-stack";
+import type { BottomTabScreenProps } from "@react-navigation/bottom-tabs";
+import type Octicons from "@expo/vector-icons/Octicons";
+import type { Credential, Profile } from "@/types/models";
+
+export type AppNavigatorInterface = {
+  WelcomeScreen: undefined;
+  CreateScreen: undefined;
+  Tabs: NavigatorScreenParams<TabNavigatorInterface>;
+  ProfileDetailScreen: { profile: Profile };
+  CredentialDetailScreen: {
+    heading: string;
+    subtitle: string;
+    iconName: keyof typeof Octicons.glyphMap;
+  };
+  AddProfileScreen: undefined;
+  AddCredentialsScreen: undefined;
+  AddCredentialDetailScreen: { credential: Credential };
```

Comment:
truthfully almost none of these params are necessary given static mocks. but defining them here makes it easier to remove them as needed.